src/NSwag.SwaggerGeneration.AspNetCore/AspNetCoreToSwaggerGenerator.cs

```diff
@@ -0,0 +1,257 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="AspNetCoreToSwaggerGenerator.cs" company="NSwag">
+//     Copyright (c) Rico Suter. All rights reserved.
+// </copyright>
+// <license>https://github.com/NSwag/NSwag/blob/master/LICENSE.md</license>
+// <author>Rico Suter, mail@rsuter.com</author>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using NJsonSchema;
+using NJsonSchema.Infrastructure;
+using NSwag.SwaggerGeneration.Processors.Contexts;
+
+namespace NSwag.SwaggerGeneration.AspNetCore
+{
+    /// <summary>Generates a <see cref="SwaggerDocument"/> object for the given Web API class type. </summary>
+    public class AspNetCoreToSwaggerGenerator
+    {
+        private readonly SwaggerJsonSchemaGenerator _schemaGenerator;
+
+        /// <summary>Initializes a new instance of the <see cref="AspNetCoreToSwaggerGenerator" /> class.</summary>
+        /// <param name="settings">The settings.</param>
+        public AspNetCoreToSwaggerGenerator(AspNetCoreToSwaggerGeneratorSettings settings)
+            : this(settings, new SwaggerJsonSchemaGenerator(settings))
+        {
+        }
+
+        /// <summary>Initializes a new instance of the <see cref="AspNetCoreToSwaggerGenerator" /> class.</summary>
+        /// <param name="settings">The settings.</param>
+        /// <param name="schemaGenerator">The schema generator.</param>
+        public AspNetCoreToSwaggerGenerator(AspNetCoreToSwaggerGeneratorSettings settings, SwaggerJsonSchemaGenerator schemaGenerator)
+        {
+            Settings = settings;
+            _schemaGenerator = schemaGenerator;
+        }
+
+        /// <summary>Gets or sets the generator settings.</summary>
+        public AspNetCoreToSwaggerGeneratorSettings Settings { get; }
+
+        /// <summary>Generates a Swagger specification for the given <see cref="ApiDescriptionGroupCollection"/>.</summary>
+        /// <param name="apiDescriptionGroups"><see cref="ApiDescriptionGroup"/>.</param>
+        /// <returns>The <see cref="SwaggerDocument" />.</returns>
+        /// <exception cref="InvalidOperationException">The operation has more than one body parameter.</exception>
+        public Task<SwaggerDocument> GenerateAsync(ApiDescriptionGroupCollection apiDescriptionGroups)
+        {
+            var apiDescriptions = apiDescriptionGroups.Items.SelectMany(g => g.Items);
+            return GenerateAsync(apiDescriptions);
```

Comment:
I think maybe this is an open question, or a future improvement but we might not want to destroy the group information. 

The intended usage of groups is for tags or versioning. See the grouping in http://petstore.swagger.io/ for example.

I'm a little concerned if we build an API that **can't** expose the groupings.

Reply:
Sure. I wasn't sure what that translated in to here. I'll add it as tags for the time being if it's present.

Reply:
I'm not trying to suggest that we just **do** something without an agreement that it's the right thing. I want to make sure that the API we build doesn't prevent it.